### ISSUE TITLE
Roadmap for documentation completion

### ISSUE THREAD

**Issue by SimonBiggs** [RadiotherapyAI/RadiotherapyAI#62]
# Document Roadmap TechDoc

Step through the completion of each of the documents below. Once complete, record the respective PR where linking to the document completion and approval. At the end of this issue is a complete list of every file for checking off after completed.

## PHASE 1: Planning & Feasibility

| Document                                        | Author                        | Comment                                                                                                                                                                            | PR Reference |
| ----------------------------------------------- | ----------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ------------ |
| Intended Use                                    | CEO                           | This is the most important document of your TechDoc. Business and Regulatory Strategy depend on it.                                                                                |              |
| Medical Device Classification                   | QA/RA Manger                  | Defining the product's medical device classification acc. to MDR.                                                                                                                  |              |
| Product Roadmap                                 | Product Manager               | Not an essential document. This is for feasibility assessments and resource planning. Helps to keep track of what's to come.                                                       |              |
| Software Development and Maintenance Plan       | Software Developer            | Giving product-specific information on the tools, resources and methods to be used for software development. Helpful for developer teams to align regarding the development setup. |              |
| Change Evaluation List                          | QA/RA Manger                  | Listing and assessing gaps between the last and the upcoming software version. Follows criteria from MDCG 2020-3. Not needed for the very first release.                           |              |
| Risk Management Plan and Risk Acceptance Matrix | Risk Manager                  | Specifies the methods for assessing risks to be used for the product at hand. Defines probability and severity categories and acceptable ranges.                                   |              |
| Clinical Evaluation Plan                        | Clinician / Scientific Person | Specifying the methodology that shall be used to assess the clinical benefits and risks of the product.                                                                            |              |

## PHASE 2: Specification

| Document                             | Author                               | Comment                                                                                                    | PR Reference |
| ------------------------------------ | ------------------------------------ | ---------------------------------------------------------------------------------------------------------- | ------------ |
| User Needs List                      | Product Manager                      | This is a list of high-level requirements. Most of them will be implemented through Software Requirements. |              |
| User Needs Checklist                 | QA/RA Manager                        | Checking if the User Needs List makes sense.                                                               |              |
| Software Requirements List           | Developer-adjacent person            | Specifying how User Needs will be incorporated in the software. Describing the details of a feature.       |              |
| List of Hazard-Related Use Scenarios | Risk Manager                         | Identifying scenarios in which users could be prone to hazards.                                            |              |
| Risk Table                           | Risk Manager                         | Anticipating and assessing risks. Gathering input from various channels.                                   |              |
| Software Requirements Checklist      | QA/RA Manager                        | Checking if the Software Requirements List makes sense.                                                    |              |
| Software Test Plan                   | Software Developer                   | Defining tests for every Software Requirement and mapping them to each other.                              |              |
| Usability Test Plan                  | Product Manager / Usability Engineer | Specifying the methodologies to be used to conduct the Usability Test.                                     |              |

## PHASE 3: Development

| Document              | Author             | Comment                                                                                                         | PR Reference |
| --------------------- | ------------------ | --------------------------------------------------------------------------------------------------------------- | ------------ |
| SOUP List             | Software Developer | List of external libraries, frameworks and packages used in the product, incl. their assessment of criticality. |              |
| Software Architecture | Software Developer | Description/Graphic outlining the software components and their interaction.                                    |              |

The actual programming work takes place in this phase. It makes sense to create the SOUP list and the software
architecture immediately before programming begins and to adjust them if anything changes during code
creation. These documents should actually be created in advance, but due to the iterative nature of software
development, it will not be possible to plan everything before the actual work begins. Therefore, it makes
sense to create them in parallel with the programming.

## PHASE 4: Verification and Validation

| Document                   | Author                               | Comment                                                                                                                                                                           | PR Reference |
| -------------------------- | ------------------------------------ | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ------------ |
| Software Test Results      | QA Tester                            | Results of the Software Tests: passed or failed?                                                                                                                                  |              |
| List of Known Anomalies    | Software Developer / Product Manager | List of known bugs and failed software tests. Justification why the software can still be released without impacting benefit/safety and a timeline when those bugs will be fixed. |              |
| Instructions For Use       | Person with the best overview        | The name says it all.                                                                                                                                                             |              |
| Usability Test Protocol    | Product Manager / Usability Engineer | Specifying the actual usability test cases: Testing User Needs, Hazard-Related Use Scenarios and Instructions for Use.                                                            |              |
| Usability Test Report      | Product Manager                      | Summary of the results of the Usability Test.                                                                                                                                     |              |
| Clinical Evaluation Report | Clinician / Scientific Person        | Analysing scientific data to prove the products safety and efficacy.                                                                                                              |              |
| Risk Management Report     | Risk Manager                         | Summary of the Risk Management Activities                                                                                                                                         |              |

## PHASE 5: Product Release

| Document                   | Author                                | Comment                                                                                                                                                             | PR Reference |
| -------------------------- | ------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ------------ |
| GSPR List                  | QA/RA Manager                         | Checking if the "General Safety and Performance Requirements" are fulfilled.                                                                                        |              |
| PMS (/PMCF) Plan           | QA/RA Manager                         | Planning the product-specific activities for Post-Market Surveillance. If the clinical data is not sufficient, also plan Post-Market Clinical Follow-Up activities. |              |
| Software Release Checklist | QA/RA Manager                         | Checking if all the requirements are fulfilled, if the documents are complete and if the product is safe to be used.                                                |              |
| Release Notes              | Product Manager OR Software Developer | Description of new features in the current release.                                                                                                                 |              |
| Declaration of Conformity  | QA/RA and CEO                         | Declaring the product's conformity with the regulations and standards.                                                                                              |              |

---

- [ ] index
- [ ] 0-intended-use/index
- [ ] 1-qms/index
- [ ] 1-qms/0-quality-manual-policy-objectives
- [ ] 1-qms/13485-mapping
- [ ] 1-qms/sop-feedback-management
- [ ] 1-qms/sop-internal-audit
- [ ] 1-qms/sop-product-certification-and-registration
- [ ] 1-qms/sop-update-of-regulations
- [ ] 1-qms/capa/0-sop
- [ ] 1-qms/capa/list-capa
- [ ] 1-qms/document-and-record-control/0-sop
- [ ] 1-qms/document-and-record-control/document-list-dlqms
- [ ] 1-qms/human-resources/0-sop
- [ ] 1-qms/human-resources/list-of-medical-device-consultants
- [ ] 1-qms/human-resources/list-of-training-documentation
- [ ] 1-qms/human-resources/tmpl-training-record
- [ ] 1-qms/management-review/0-sop
- [ ] 1-qms/management-review/management-review-report
- [ ] 1-qms/post-market-surveillance/0-sop
- [ ] 1-qms/post-market-surveillance/periodic-safety-update-report
- [ ] 1-qms/post-market-surveillance/post-market-clinical-follow-up-plan
- [ ] 1-qms/post-market-surveillance/post-market-clinical-follow-up-report
- [ ] 1-qms/post-market-surveillance/post-market-surveillance-plan
- [ ] 1-qms/purchasing/0-sop
- [ ] 1-qms/purchasing/list-of-qualified-suppliers
- [ ] 1-qms/purchasing/supplier-checklist
- [ ] 1-qms/software-validation/0-sop
- [ ] 1-qms/software-validation/software-list
- [ ] 1-qms/software-validation/software-validation-form
- [ ] 1-qms/vigilance/0-sop
- [ ] 1-qms/vigilance/tmpl-field-safety-notice
- [ ] 1-qms/vigilance/tmpl-incident-assessment-form
- [ ] 2-techdoc/index
- [ ] 2-techdoc/document-roadmap
- [ ] 2-techdoc/list-of-known-anomalies
- [ ] 2-techdoc/14971/index
- [ ] 2-techdoc/14971/risk-acceptance-matrix
- [ ] 2-techdoc/14971/risk-management-plan
- [ ] 2-techdoc/14971/risk-management-report
- [ ] 2-techdoc/14971/risk-table-fmea/index
- [ ] 2-techdoc/62304/index
- [ ] 2-techdoc/62304/62304-mapping
- [ ] 2-techdoc/62304/algorithm-validation-report
- [ ] 2-techdoc/62304/bug-fixes-documentation-list
- [ ] 2-techdoc/62304/change-evaluation-list
- [ ] 2-techdoc/62304/checklist-software-release
- [ ] 2-techdoc/62304/checklist-software-requirements-review
- [ ] 2-techdoc/62304/checklist-user-needs-review
- [ ] 2-techdoc/62304/list-of-medical-devices
- [ ] 2-techdoc/62304/software-architecture-checklist
- [ ] 2-techdoc/62304/software-architecture-description
- [ ] 2-techdoc/62304/software-development-maintenance-plan
- [ ] 2-techdoc/62304/software-requirements-list
- [ ] 2-techdoc/62304/software-system-test-plan
- [ ] 2-techdoc/62304/sop-change-management
- [ ] 2-techdoc/62304/sop-deployment
- [ ] 2-techdoc/62304/sop-integrated-software-development
- [ ] 2-techdoc/62304/sop-software-problem-resolution
- [ ] 2-techdoc/62304/soup-list
- [ ] 2-techdoc/62304/user-needs-list
- [ ] 2-techdoc/62366/index
- [ ] 2-techdoc/62366/62366-mapping
- [ ] 2-techdoc/62366/list-of-hazard-related-use-scenarios
- [ ] 2-techdoc/62366/usability-evaluation-plan
- [ ] 2-techdoc/62366/usability-evaluation-protocol
- [ ] 2-techdoc/62366/usability-evaluation-report
- [ ] 3-general/index
- [ ] 3-general/checklist-mdr
- [ ] 3-general/declaration-of-conformity-mdd-class-i
- [ ] 3-general/essential-requirements-mdd
- [ ] 3-general/instructions-for-use
- [ ] 3-general/mdd-classification-document
- [ ] 3-general/mdr-classification-document
- [ ] clinical-evaluation/0-sop-clinical-evaluation
- [ ] clinical-evaluation/clinical-evaluation-plan
- [ ] clinical-evaluation/clinical-evaluation-report
- [ ] data-protection/index
- [ ] data-protection/data-protection-statement-website
- [ ] data-protection/employee-data-protection-and-confidentiality-agreement
- [ ] data-protection/list-of-data-processing-activities
- [ ] data-protection/technical-and-organizational-measures
